### PR TITLE
docs: improve development documentation

### PR DESCRIPTION
**Pull request by christian-bromann**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally for affected output targets
- [x] Tests (`npm test`) were run locally and passed
- [x] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

With the recent changes to Vue and React output targets some things have changed that haven't been documented. This patch updates the development docs of this project.

## What is the new behavior?

- consolidate development commands, e.g. `dev`, `test.watch`, `test.unit`
- document development commands
- add more code documentations
- fix type issues with Vue SSR implementation

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,6 +24,13 @@ At this time, only new issues & pull requests that support the Ionic Framework w
 
 This project uses [pnpm](https://pnpm.io/) for package management. You can install it by following the [installation instructions](https://pnpm.io/installation).
 
+## Requirements
+
+This project requires:
+
+- Node.js v20.10.0 or higher.
+- [pnpm](https://pnpm.io/) v9.4.0 or higher
```
Comment (tanner-reits) [inline README.md:32]: We gonna actually remember to update this if Renovate changes the engine values?

**Review by tanner-reits — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (christian-bromann) [reply README.md:32]: I was thinking of adding it to the `ignoreDeps` list but then this is not a dependency and rather an version specification for an engine so I believe we won't see it popping up.

Comment (tanner-reits) [reply README.md:32]: Renovate will change the engine config. Here's an example from Stencil Core https://github.com/ionic-team/stencil/pull/5791

Comment (christian-bromann) [reply README.md:32]: I tried ignoring the engine fields and Renovate's validator said that pnpm is not a supported package manager, let's see what happens.

**Pull request merged**